cmd/wharf/config.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"os"
+
+	"github.com/iver-wharf/wharf-cmd/pkg/aggregator"
+	"github.com/iver-wharf/wharf-cmd/pkg/provisioner"
+	"github.com/iver-wharf/wharf-cmd/pkg/provisionerapi"
+	"github.com/iver-wharf/wharf-cmd/pkg/worker"
+	"github.com/iver-wharf/wharf-core/v2/pkg/config"
+)
+
+// Config holds all configurable settings for wharf-api.
+//
+// The config is read in the following order:
+//
+// 1. File: ~/.config/iver-wharf/wharf-cmd/wharf-cmd-config.yml
+//
+// 2. File: ./wharf-cmd-config.yml
+//
+// 3. File from environment variable: WHARF_CMD_CONFIG
+//
+// 4. Environment variables, prefixed with WHARF_CMD
```

Comment:
Suggest just using `WHARF_CONFIG` and `WHARF_CMD_` to stay consistent with the other repos. No need to scope it to a project level IMO

Reply:
Up for discussion though. What do you think?

Reply:
I think it would be nice to scope it to future-proof a bit against collisions.
Also makes it more obvious which part of Wharf it relates to when seeing the name.

I changed it back to not have `_CMD` in it for now, but makes more sense to me to have it in.

Open for discussion too 👍🏻

Reply:
I'm mostly worried if we go with `WHARF_CMD_` here without updating the others then we'd have:

- wharf-cmd: `WHARF_CMD_`
- wharf-api, wharf-provisioner-github, etc: `WHARF_`

Which is a confusing to use.

Due to a technical limitation, we cannot just add two environment variables sources via configs (see https://pkg.go.dev/github.com/iver-wharf/wharf-core/pkg/config#Builder), so there'd be no easy way of adding backward compatiblity if we change all to have component-speicfic prefixes.

Reply:
Hmm, I see.
In that case I'll continue with just `WHARF_`.